### ISSUE TITLE
igx-card-header should have no role attribute

### ISSUE THREAD

**Issue by kdinev** [IgniteUI/igniteui-angular#12792]
## Description  
The `igx-card-header` elements has `role="header"` attribute. `header` is an invalid role. The header element is a semantic header, which is implied by the tag name and there should be no role.

 * igniteui-angular version: 15.0.x
 * browser: all

## Steps to reproduce  

1. Open any card sample
2. Inspect the header element
3. ...

## Result  
The header element has an invalid role.

## Expected result  
The header element should have no role.

## Attachments  
![image](https://user-images.githubusercontent.com/1472513/226353402-504dfd21-3a03-436a-bdd1-558d93d89754.png)
